### PR TITLE
feat: Add glob patterns to match manager files

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Regexes are powerful, but error-prone
- They're hard to write inside string literals
- This is the first step for porting `fileMatch` to more user-friendly glob syntax
- It doesn't change existing Renovate behavior

## Context

- Ref: #21737

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gradle/index.ts [hunk 1]
```diff
@@ -20,6 +20,17 @@ export const defaultConfig = {
...
+    '**/*.gradle',
+    '**/*.gradle.kts',
+    '**/gradle.properties',
+    '**/gradle/**/*.toml',
+    '**/buildSrc/**/*.kt',
+    '**/*.versions.toml',
+    // The two below is for gradle-consistent-versions plugin
+    '**/versions.props',
+    '**/versions.lock',
+  ],
```
Comment (zharinov) [inline lib/modules/manager/gradle/index.ts:33]: @Churro Are these glob patterns for Gradle correct?

**Comment by rarkins** [comment]
I don't think we should have both at the same time

**Comment by zharinov** [comment]
In this case, it's the draft

**Comment by rarkins** [comment]
Let's have this field support regex too, denoted by /.../

**Comment by rarkins** [comment]
New ref: https://github.com/renovatebot/renovate/issues/10582

File: lib/modules/manager/azure-pipelines/index.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export { extractPackageFile } from './extract';
 
 export const defaultConfig = {
   fileMatch: ['azure.*pipelines?.*\\.ya?ml$'],
+  filePatterns: ['**/azure-pipelines.{yml,yaml}'], // not used yet
```
Comment (rarkins) [inline lib/modules/manager/azure-pipelines/index.ts:7]: This one is not the same in two ways

File: lib/modules/manager/bicep/index.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export { extractPackageFile } from './extract';
 
 export const defaultConfig = {
   fileMatch: ['\\.bicep$'],
+  filePatterns: ['**/*.bicep'], // not used yet
```
Comment (rarkins) [inline lib/modules/manager/bicep/index.ts:7]: Won't match `.bicep` as a filename but not sure if that's a valid name anyway or if it's meant to be like `xyz.bicep`

File: lib/modules/manager/docker-compose/index.ts [hunk 4]
```diff
@@ -8,6 +8,15 @@ export { extractPackageFile };
...
   fileMatch: ['(^|/)(?:docker-)?compose[^/]*\\.ya?ml$'],
+  filePatterns: [
+    '**/compose.{yml,yaml}',
+    '**/compose-*.{yml,yaml}',
+    '**/compose.*.{yml,yaml}',
+    '**/compose_*.{yml,yaml}',
+    '**/docker-compose.{yml,yaml}',
+    '**/docker-compose.*.{yml,yaml}',
+    '**/docker-compose-*.{yml,yaml}',
+  ], // not used yet
```
Comment (rarkins) [inline lib/modules/manager/docker-compose/index.ts:19]: This ain't great

**Comment by rarkins** [comment]
Idea: make make `filePatterns` mergeable. If we have default patterns which users don't want, then they should use `ignorePaths` instead.

**Review by viceice — COMMENTED** [review]
conflicted

**Comment by rarkins** [comment]
Related: https://github.com/renovatebot/renovate/pull/24695/files

Let's reuse this branch later.

**Pull request closed without merging**